jina/peapods/pods/k8s.py

```diff
@@ -256,6 +256,63 @@ async def wait_restart_success(self, previous_uids: Iterable[str] = None):
                 fail_msg += f': {repr(exception_to_raise)}'
             raise RuntimeFailToStart(fail_msg)
 
+        async def wait_scale_success(self, replicas: int):
+            scale_to = replicas
+            _timeout = self.common_args.timeout_ready
+            if _timeout <= 0:
+                _timeout = None
+            else:
+                _timeout /= 1e3
+
+            import asyncio
+            from kubernetes import client
+
+            with JinaLogger(f'waiting_scale_for_{self.name}') as logger:
+                logger.info(
+                    f'🏝️\n\t\tWaiting for "{self.name}" to be scaled, with {self.num_replicas} replicas,'
+                    f'scale to {scale_to}.'
+                )
+                timeout_ns = 1000000000 * _timeout if _timeout else None
+                now = time.time_ns()
+                exception_to_raise = None
+                while timeout_ns is None or time.time_ns() - now < timeout_ns:
+                    try:
+                        api_response = kubernetes_client.K8sClients().apps_v1.read_namespaced_deployment_scale(
```

Comment:
are we sure that `status.replicas` in the response actually contains only the ready replicas?